### ISSUE TITLE
Watch for Tesla

### ISSUE THREAD

**Issue by tim7win** [Kemmey/Kemmey-TeslaWatch-Public#203]
NOT SO COOL APP for Apple Watch 6 The Logon   Crashes any and All Tesla App TEZLAB STATS
After Loging on I get the following message: Couldnot acquire Token: Request FAILED RESPONSE CODE : 401 MESSAGE: UNAUTHORIZED

**Comment by Kemmey** [comment]
Yeah, that's an ongoing issue. Please see this thread: #190

The solution is to login with the new mfa system, which can only be done in a browser, that the watch does not have. So I've added an iOS counterpart. You can join the beta and get access to that pre-release version now, by joining here: https://testflight.apple.com/join/FdzFOLZv

**Comment by tim7win** [comment]
Thanks for the Information, but It is above my pay grade to test this 

Out I will have to wait till the problem is resolved.

**Comment by Kemmey** [comment]
@tim7win One thing - did you already try logging in on Tesla.com and verify your account was not locked out from too many failed login attempts?

**Comment by tim7win** [comment]
Yes it was locked because of the attempts. I reset Tesla apps but it still requires a API verification code and I am not receiving one in texts or emails

**Comment by tim7win** [comment]
I know thatTesla .coo doesn’t require a API does the watch for tesla require one

**Comment by Kemmey** [comment]
@tim7win I'm not sure I follow. 

When you say it require an api code, is this you citing the error message you see when trying to log in? That code is an error number. 401 means access denied, basically that either your password has a typo, or your account got locked out. Tesla locks out your account after only two attempts with a typo in the password, so it can easily happen. 

For the next release, I'm adding a phone app as well, which will allow you to do the login on the phone, instead of fiddling with the watch.

**Comment by tim7win** [comment]
When I would log into the watch for tesla it asks for my user i’d and password I then click login on my watch it acknowledges my log on in and request for the verification code which I do not receive  I then scroll the crown and I see the error message that I sent you All my programs that require a verification code I receive it in a text message to be entered and then submit to open the app

**Comment by Kemmey** [comment]
@tim7win The watch app does not request a verification code. I would know, I wrote the authentication flow. What you're seeing is an error code. That error code means your password was wrong or your account was locked. 

Even with multi factor authentication turned on, the old authentication endpoints still work without multi factor. That is the entire reason for the new version. To use the new multi factor auth flow. But that is currently not necessary, as the old works just fine. 

For most people. But what you're seeing is not the error the affected people are seeing. They can successfully login just fine. Only their login expires within a few hours.

**Comment by tim7win** [comment]
After 30 attempts I was finally successfull in logging in  Thanks for all you help have a FUNderfull weekend
